File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Stack<Integer> stack = new Stack<>();
		int h;
		int n = Integer.parseInt(br.readLine());
		long sum = 0;

		for (int i = 0; i < n; i++) {
			h = Integer.parseInt(br.readLine());
			while (!stack.isEmpty()) {
				if (stack.peek() <= h)
					stack.pop();
				else
					break;
			}
			sum += stack.size();
			stack.push(h);
		}
		System.out.println(sum);
	}
}

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.cc
#include <iostream>
#include <stack>
using namespace std;
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  int N, h;
  long long sum = 0;
  cin >> N;
  stack<int> s;
  while (N--) {
    cin >> h;
    while (!s.empty() && s.top() <= h) s.pop();
    sum += s.size();
    s.push(h);
  }
  cout << sum;
  return 0;
}